source/common/filter/config_discovery_impl.h

```diff
@@ -387,9 +392,42 @@ class FilterConfigProviderManagerImplBase : Logger::Loggable<Logger::Id::filter>
                                          absl::string_view type_url) const;
   void validateProtoConfigTypeUrl(const std::string& type_url,
                                   const absl::flat_hash_set<std::string>& require_type_urls) const;
+  virtual const std::string getConfigDumpType() const PURE;
 
 private:
+  void setupEcdsConfigDumpCallbacks(Server::Admin& admin) {
+    if (config_tracker_entry_ == nullptr) {
+      config_tracker_entry_ = admin.getConfigTracker().add(
+          getConfigDumpType(), [this](const Matchers::StringMatcher& name_matcher) {
```

Comment:
the config dump callbacks are stored in a keyed hashmap, thus the keys has to be different to avoid one override the other. Thus we need getConfigDumpType() to get different strings for each filter type, like HTTP filter, TCP listener filter,...